bare-metal/system_timer.c

```diff
@@ -35,19 +36,54 @@ typedef struct sys_timer_struct_s {
     ns_list_link_t link;
 } sys_timer_struct_s;
 
+#define TIMER_SLOTS_PER_MS          20
 #define TIMER_SYS_TICK_PERIOD       10 // milliseconds
-#define TIMER_SYS_TICK_SLOTS        (TIMER_SYS_TICK_PERIOD * 20) // 50us slots
+#define TIMER_SYS_TICK_SLOTS        (TIMER_SYS_TICK_PERIOD * TIMER_SLOTS_PER_SECOND) // 50us slots
 
 static uint32_t run_time_tick_ticks = 0;
 static NS_LIST_DEFINE(system_timer_free, sys_timer_struct_s, link);
 static NS_LIST_DEFINE(system_timer_list, sys_timer_struct_s, link);
-static int8_t sys_timer_id = -1;
 
 
 static sys_timer_struct_s *sys_timer_dynamically_allocate(void);
-static void timer_sys_interrupt(int8_t timer_id, uint16_t slots);
+static void timer_sys_interrupt(void);
 
+#ifndef NS_EVENTLOOP_USE_TICK_TIMER
+int8_t platform_tick_timer_start(uint32_t milliseconds);
```

Comment:
@kjbracey-arm I think this has to be here because in this case this is not declared in the header. Or if you prefer moving the implementation before the callback function?